### ISSUE TITLE
`SlevomatCodingStandard.TypeHints.PropertyTypeHint` false positive when `WeakMap<static, object{a: int}>` (object shape). 

### ISSUE THREAD

**Issue by LastDragon-ru** [slevomat/coding-standard#1692]
```php
<?php declare(strict_types = 1);

namespace Bug;

use WeakMap;

class Bug {
    /**
     * @var WeakMap<static, object{a: int}> https://phpstan.org/writing-php-code/phpdoc-types#object-shapes
     */
    public WeakMap $objectMap;
    /**
     * @var WeakMap<static, array{a: int}>
     */
    public WeakMap $arrayMap;
}
```

```xml
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor-bin/phpcs/vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <config name="installed_paths" value="../../slevomat/coding-standard"/>
    <rule ref="SlevomatCodingStandard.TypeHints.PropertyTypeHint"/>
</ruleset>
```

Actual:

```
--------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------
 9 | ERROR | @var annotation of property \Bug\Bug::$objectMap does not specify type hint for its items.
--------------------------------------------------------------------------------------------------------
```

Expected:

```
no error
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/3c0ae902c48447cf43690101c1cd4d1e3ff3ae54
